vita3k/gui/src/settings_dialog.cpp

```diff
@@ -89,16 +93,40 @@ static bool change_pref_location(const std::string &input_path, const std::strin
     return true;
 }
 
-static bool clear_and_refresh_game_list(HostState &host, GuiState &gui) {
-    if (gui.game_selector.games.empty())
-        return false;
+static void change_emulator_path(GuiState &gui, HostState &host) {
+    nfdchar_t *emulator_path = nullptr;
+    nfdresult_t result = NFD_PickFolder(nullptr, &emulator_path);
 
-    gui.game_selector.games.clear();
-    if (!gui.game_selector.games.empty())
-        return false;
+    if (result == NFD_OKAY && emulator_path != host.pref_path) {
+        // Refresh the working paths
+        host.cfg.pref_path = static_cast<std::string>(emulator_path) + '\\';
+        host.pref_path = host.cfg.pref_path;
 
-    get_game_titles(gui, host);
-    return true;
+        config::serialize_config(host.cfg, host.cfg.config_path);
+        LOG_INFO_IF(refresh_game_list(gui, host), "Successfully moved Vita3K path to: {}", host.pref_path);
+    }
+}
+
+static bool change_user_image_background(GuiState &gui, HostState &host) {
+    nfdchar_t *image_path = nullptr;
+    nfdresult_t result = NFD_OpenDialog("bmp,gif,jpg,png,tif", nullptr, &image_path);
+
+    if (result == NFD_OKAY && host.cfg.background_image != static_cast<std::string>(image_path)) {
+        const std::string image_path_str = static_cast<std::string>(image_path);
+
+        if (!gui.user_backgrounds[image_path_str])
+            init_background(gui, image_path_str);
+        else if (gui.user_backgrounds[image_path_str])
+            gui.current_background = gui.user_backgrounds[image_path_str];
+
+        if (gui.user_backgrounds[image_path_str]) {
+            host.cfg.background_image = image_path_str;
+            return true;
+        } else
+            return false;
```

Comment:
You can merge this with the above if statement. (no need to check if image_path_str is in user_backgrounds twice).

Reply:
I have set all this for good raison

Reply:
Hmm... Can you make init_background return a bool so we can check if it failed there instead of checking with fs::exists again? This looks more or less like a duplicate statement.